+nosnoc/+solver/MpccSolver.m

```diff
@@ -426,5 +458,415 @@ function print_infeasibility(obj, results)
+                    ublift_H(find(ind_00 | ind_p0)) = 0;
+
+                    G_init = G_old;
+                    G_init(ind_00 | ind_0p) = 0;
+                    H_init = H_old;
+                    H_init(ind_00 | ind_p0) = 0;
+
+                    lbw = vertcat(lbw_orig, lblift_G, lblift_H);
+                    ubw = vertcat(ubw_orig, ublift_G, ublift_H);
+                    w_init = vertcat(w_orig, G_init, H_init);
+
+                    
+                    casadi_nlp = struct('f', f , 'x', w, 'g', g, 'p', p);
+                    opts_casadi_nlp.ipopt.max_iter = 5000;
+                    opts_casadi_nlp.ipopt.warm_start_init_point = 'yes';
+                    opts_casadi_nlp.ipopt.warm_start_entire_iterate = 'yes';
+                    opts_casadi_nlp.ipopt.warm_start_bound_push = 1e-5;
+                    opts_casadi_nlp.ipopt.warm_start_bound_frac = 1e-5;
+                    opts_casadi_nlp.ipopt.warm_start_mult_bound_push = 1e-5;
+                    opts_casadi_nlp.ipopt.tol = obj.opts.comp_tol;
+                    opts_casadi_nlp.ipopt.acceptable_tol = sqrt(obj.opts.comp_tol);
+                    opts_casadi_nlp.ipopt.acceptable_dual_inf_tol = sqrt(obj.opts.comp_tol);
+                    opts_casadi_nlp.ipopt.fixed_variable_treatment = 'relax_bounds';
+                    opts_casadi_nlp.ipopt.honor_original_bounds = 'yes';
+                    opts_casadi_nlp.ipopt.bound_relax_factor = 1e-12;
+                    opts_casadi_nlp.ipopt.linear_solver = obj.opts.opts_casadi_nlp.ipopt.linear_solver;
+                    opts_casadi_nlp.ipopt.mu_strategy = 'adaptive';
+                    opts_casadi_nlp.ipopt.mu_oracle = 'quality-function';
+                    opts_casadi_nlp.ipopt.nlp_scaling_method = 'none';
+                    default_tol = 1e-7;
+                    opts_casadi_nlp.ipopt.tol = default_tol;
+                    opts_casadi_nlp.ipopt.dual_inf_tol = default_tol;
+                    opts_casadi_nlp.ipopt.dual_inf_tol = default_tol;
+                    opts_casadi_nlp.ipopt.compl_inf_tol = default_tol;
+                    opts_casadi_nlp.ipopt.resto_failure_feasibility_threshold = 0;
+                    opts_casadi_nlp.ipopt.print_level = 5;
+                    opts_casadi_nlp.ipopt.sb = 'yes';
+                    
+                    tnlp_solver = nlpsol('tnlp', 'ipopt', casadi_nlp, opts_casadi_nlp);
+                    tnlp_results = tnlp_solver('x0', w_init,...
+                        'lbx', lbw,...
+                        'ubx', ubw,...
+                        'lbg', lbg,...
+                        'ubg', ubg,...
+                        'lam_g0', lam_g_aug,...
+                        'lam_x0', lam_x_aug,...
+                        'p', nlp.p.mpcc_p().val);
+                    res_out = tnlp_results;
+                    if tnlp_solver.stats.success
+                        converged = true;
+                    else
+                        n_biactive = n_biactive + 1;
```

Comment:
?

Reply:
This is part of the algorithm that adds more (possibly ambiguous) points to the biactive set. In some cases the initial guess at a biactive set may be inaccurate and some ambiguous points are also biactive (this for example often happens in time freezing problems). As such we sequentially add points to the biactive set and try to solve the TNLP again.

Reply:
I see, a description in the comment would be fine. 
And when this procedure stops should be explained.